File: 807.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        
        x = [0] * m;
        y = [0] * n;
        
        for i in range(0, m):
            for j in range(0, n):
                x[i] = max(x[i], grid[i][j])
                y[j] = max(y[j], grid[i][j])
        res = 0;
        
        for i in range(0, m):
            for j in range(0, n):
                res += min(x[i], y[j]) - grid[i][j]
        return res

File: 807.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        vector<int>x(m, 0);
        vector<int>y(n, 0);
        for(int i = 0; i < m; i++)
            for(int j = 0; j < n; j++){
                x[i] = max(x[i], grid[i][j]);
                y[j] = max(y[j], grid[i][j]);
            }
        int res = 0;
        for(int i = 0; i < m; i++)
            for(int j = 0; j < n; j++)
                res += min(x[i], y[j]) - grid[i][j];
        
        return res;
    }
};
